### PR TITLE
feat: upgrade react to version 18 with typings

### PR DESCRIPTION
**Pull request by yoannfleurydev**
From 164 typings issue to 11 at the moment

### CODE REVIEW

**Comment by Jessy-BAER** [comment]
Down to 0

File: tsconfig.json [hunk 1]
```diff
@@ -23,5 +23,9 @@
     "typeRoots": ["node_modules/@types", "src/types"]
   },
   "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
-  "exclude": ["node_modules"]
+  "exclude": [
+    "node_modules",
+    "/src/components/ConfirmPopover/index.tsx",
+    "src/components/ConfirmPopover/index.tsx"
```
Comment (yoannfleurydev) [inline tsconfig.json:29]: If we are not doing the Chakra update now, I suggest to add a comment to remove those lines.

**Comment by yoannfleurydev** [comment]
> Down to 0

bitmoji

File: tsconfig.json (hunk 1 above)
Comment (Jessy-BAER) [reply tsconfig.json:29]: this is a mistake

**Comment by yoannfleurydev** [comment]
Also, quick note so we do not forget this comment https://github.com/BearStudio/start-ui-web/commit/736dee6acb622016fa72d5ca0d29038c1169a8bd#r74286544

**Review by DecampsRenan — APPROVED** [review]

**Review by HugoPerard — APPROVED** [review]
Maybe you can add a custom type like that :
`FCWithChildren<T> = FC<React.PropsWithChildren<T = unknown>>`
Other than that question, the PR is good for me

**Pull request merged**